compiler/rustc_mir_dataflow/src/debuginfo.rs

```diff
@@ -0,0 +1,20 @@
+use rustc_index::bit_set::BitSet;
+use rustc_middle::mir::visit::*;
+use rustc_middle::mir::*;
+
+/// Return the set of locals that appear in debuginfo.
+pub fn debuginfo_locals(body: &Body<'_>) -> BitSet<Local> {
```

Comment:
Nit: it may be slightly more error-proof to put the debug-info level check into this function, that way it won't be forgotten in the future

Reply:
I'm not convinced. Having `debuginfo_locals` return exactly the locals in debuginfo is the least surprising behaviour. Up to the caller to know what to do with it.